tools/lib/h5tools.c

```diff
@@ -597,6 +597,9 @@ h5tools_set_fapl_vfd(hid_t fapl_id, h5tools_vfd_info_t *vfd_info)
                     if (H5Pset_fapl_subfiling(fapl_id, (const H5FD_subfiling_config_t *)vfd_info->info) < 0)
                         H5TOOLS_GOTO_ERROR(FAIL, "H5Pset_fapl_subfiling() failed");
                 }
+                else {
+                    H5TOOLS_GOTO_ERROR(FAIL, "MPI is not available for Subfiling VFD");
```

Comment:
See my comment on #3880; this may not be needed if we decide to remove this check.